react-common/components/share/ShareInfo.tsx

```diff
@@ -100,21 +104,54 @@ export const ShareInfo = (props: ShareInfoProps) => {
         setCopySuccessful(false);
     }
 
+    const handleKioskSubmitBlur = () => {
+        setKioskSubmitSuccessful(false);
+    }
+
+    const handleKioskSubmitClick = async () => {
+        const gameId = /([^\/]+$)/.exec(shareData.url)?.[1];
```

Comment:
I don't know reg expressions well, but why do you take `[1]` here instead of `[0]`?

Reply:
I honestly have no clue why I decided to use 1 here. I can change it to 0. I'm also not all that great with regex, but `.exec()` returns an array whose first entry is the text that matches the given regular expression and the second entry holds the first string that fit defined capturing groups. Since I wrapped the regex in a capturing group, the entries at index 0 and 1 are the same. I'm not sure if there's a benefit to referencing one over the other. I can also be missing something because I do have limited understanding and experience with regex.

Reply:
https://regex101.com/r/lSUeQc/1 is a helpful site for evaluating regexes -- the `[1]` here grabs the capture group (stuff in parentheses) rather than the match (portion of the string matching the entire regex), but in this case those two will be the same thing as it's just grabbing everything after the last `/` (that is, if you wanted you could remove the parentheses and just grab `[0]`, but again, it really doesn't matter~).

That said, we should just use `pxt.Cloud.parseScriptId` here to extract shareid~